Logging implementation on the microservices

ayushgautam opened the pull request:
Changes:
1. Log.java file: addded new class to create a filewriter and writeFIle functionality
2. Individual services:
- Initialize Log
- Call Write File method in Log class
- passed in extra username parameter from the MSClientAPI to the individual service method calls for tracking each user's actions in case of multiple sessions
3. Dockerfiles:
- Each individual service is given the Log.java file via dockerfile
4. stack-ms.yml:
- mapped directory in docker system to local system directory
5. ./ms_logs:
- Directory where log file would get generated, file write happens in append mode

ayushgautam committed:
Logging implementation on the microservices

Changes:
1. Log.java file: addded new class to create a filewriter and writeFIle functionality
2. Individual services:
- Initialize Log
- Call Write File method in Log class
- passed in extra username parameter from the MSClientAPI to the individual service method calls for tracking each user's actions in case of multiple sessions
3. Dockerfiles:
- Each individual service is given the Log.java file via dockerfile
4. stack-ms.yml:
- mapped directory in docker system to local system directory
5. ./ms_logs:
- Directory where log file would get generated, file write happens in append mode

alecyuverymuch commented on ms/ms.iml:
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
This is an Intellij file? We should remove this

ayushgautam replied:
Done

alecyuverymuch commented on ms_logs/logs.txt:
```diff
@@ -0,0 +1,8 @@
+Mon Feb 21 02:24:44 UTC 2022 | User signed up:Alok | USERNAME:N/A
```
If we are writing to the server, why do we have a log file here?

ayushgautam replied:
Removed

achat2 commented on ms_logs/logs.txt:
```diff
@@ -0,0 +1,8 @@
+Mon Feb 21 02:24:44 UTC 2022 | User signed up:Alok | USERNAME:N/A
```
Please keep `ms_logs` in .gitignore

ayushgautam replied:
Removed from the checkin

achat2 commented on ms/ms.iml:
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
This also should go in .gitignore

ayushgautam replied:
removed

achat2 commented on ms/Log.java:
```diff
@@ -0,0 +1,56 @@
+        {
+            //creating the log file
+            File myObj = new File("/logs/logs.txt");
+            if (myObj.createNewFile()) {
```
`createNewFile` would need exception handling. https://docs.oracle.com/javase/7/docs/api/java/io/File.html#createNewFile()

ayushgautam replied:
So there is already a try catch block around that. Can we talk more on this ?

achat2 commented on ms/AuthServices.java:
```diff
@@ -40,15 +44,23 @@ public static void main(String args[])
 
             System.out.println("AuthServices binding err: " + e.getMessage()); 
             e.printStackTrace();
+            if(logEnable)
```
Would it be better to put `logEnable` in Log.java instead of here?

ayushgautam replied:
I feel it makes sense for each microservice to have their own flag of log enable and disable. Decoupling each service's functionality. Open for discussion.

achat2 reviewed (changes requested):
Some comments to give general structure. You can choose to address/ignore whichever you want. 😄

ayushgautam committed:
Changes for PR comments

1. Removed files not part of the codebase

ayushgautam committed:
Added Log service name to the logs

New constructor in Log class to make logging service name visible in the log file

ayushgautam committed:
Removed filewriter from the individual services
